### DISCUSSION TITLE
Create a New Repository - "Add a README file" - Auto generated readme is missing the trailing newline

### DISCUSSION THREAD

**Discussion by tedmiston** [community/community#21160] [Other Feature Feedback, Questions, & Ideas]
When creating a new repo via https://github.com/new, there is an option: `Initialize this repository with > Add a README file` ([repo docs][0] / [readme docs][1]).

When checking this option, the repo gets created with a `README.md` that has a single line like:

```
# <repo name>
```

This file *does not* have a trailing newline.

However, on *nix operating systems, this is not actually a valid file because a line is defined as terminating with a newline.  In other words, the lack of a trailing newline in the generated file makes it invalid.

For more info, see these references:

- https://stackoverflow.com/q/729692/149428
- https://unix.stackexchange.com/q/18743/169772

Please update the generation to include a trailing newline so it is a valid file on *nix OSes!

[0]: https://docs.github.com/en/get-started/quickstart/create-a-repo
[1]: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes

**Comment by radar-blip** [comment]
Using Linux Mint. If I add a description to repo, there is a trailing newline. With no description, no newline.
Dont know if this helps, thats just what happened when I tried it.

**Comment by thuongtruong1009** [comment]
Hi @tedmiston 
* You can use [editorconfig](https://editorconfig.org/) combine with [prettier](https://prettier.io/) or [eslint](https://eslint.org/) to format code before commit to GitHub.
* In editor config  conventions have one property is a ```insert_final_newline = true```
